### PR TITLE
Mail log for #17491 and some improvements on log errors

### PR DESCRIPTION
**Pull request by mckaygerhard**
## Improve log error handling regarding notification sending for issue#17491

* Fix  problem where admins at test install cannot see the log of errors for UI test mail button, we can just see that this is the correct form cos other parts of the code manage the exception inside the catch using log interface class, we can just see that this is the correct form cos other parts of the code manage the exception inside the catch using log interface class
* when an error is generated when denying checkouts, there are not enough logs
to determine the problem from the email provider
* missing handling of log test mail config, there is none of logs cos there
is no log handling on test email, because all the results are just sent to
the http response and no log were written.

Partially addresses #17491

### CODE REVIEW

File: app/Http/Controllers/Account/AcceptanceController.php [hunk 1]
```diff
@@ -347,6 +347,7 @@ public function store(Request $request, $id) : RedirectResponse
 
             $acceptance->decline($sig_filename, $request->input('note'));
             $acceptance->notify(new AcceptanceAssetDeclinedNotification($data));
+            Log::debug('New event acceptance.');
```
Comment (snipe) [inline app/Http/Controllers/Account/AcceptanceController.php:350]: I don't think this is really needed here

File: app/Http/Controllers/Api/SettingsController.php [hunk 2]
```diff
@@ -150,8 +150,11 @@ public function ajaxTestEmail() : JsonResponse
         if (!config('app.lock_passwords')) {
             try {
                 Notification::send(Setting::first(), new MailTest());
+                Log::debug('Attempting to sending to '.config('mail.reply_to.address'));
```
Comment (snipe) [inline app/Http/Controllers/Api/SettingsController.php:153]: This one either - we don't need to know about it is it didn't trip the catch exception handler.

File: app/Http/Controllers/Account/AcceptanceController.php (hunk 1 above)
Comment (mckaygerhard) [reply app/Http/Controllers/Account/AcceptanceController.php:350]: ok, i will remove it, BUT before remove it, i try to explain and you just said if are valid or not:

when I decline a consumable device, you've confirmed that a 500 error occurs if the mail system is not already configured, this is already known, but there's no clear reason for it, i've already reviewed the issues and noticed that they don't handle the error from the response mail system.

these lines are more of a marker to identify what was happening before this problem occurred.

sop these lines will help others to find a way to solve this problem, currently i am made testing but i need this previously there for!

File: app/Http/Controllers/Api/SettingsController.php (hunk 2 above)
Comment (mckaygerhard) [reply app/Http/Controllers/Api/SettingsController.php:153]: In this case it is much more necessary, as a mark is made where we know the previous step before the problem.

This helps the less experienced to collaborate in the code since as i have confirmed the exception is not always triggered.

U guys take the email response as a status absolute reponse, means you dont manage and understand at all those, these are like 550 which does not mean an error but a warning, and this has variants like 550 5.4.5 which is a full tray, but if you send this to HTTP (here is a controller api) means will be interpreted as http error (http internal server error, variant of error 500) as ican see in the picture of the issue reported

File: app/Http/Controllers/Account/AcceptanceController.php (hunk 1 above)
Comment (mckaygerhard) [reply app/Http/Controllers/Account/AcceptanceController.php:350]: @snipe  also i see more lines like this in the LDAP part code! Soi i just followed the same practices.

File: app/Http/Controllers/Api/SettingsController.php (hunk 2 above)
Comment (snipe) [reply app/Http/Controllers/Api/SettingsController.php:153]: It really isn't. If no exception is being thrown, there's no information to be gained from that. I'll allow it in this case, but we don't normally like to junk up the logs too much. If log files are not writeable for some reason (which happens quite a lot with our users) it can break things in very unexpected ways.

I promise you don't need to explain how mail servers or http status codes work to us.

Comment (mckaygerhard) [reply app/Http/Controllers/Api/SettingsController.php:153]: Ok got it

Comment (mckaygerhard) [reply app/Http/Controllers/Api/SettingsController.php:153]: That why i used DEBUG level here.. take note will only be filled in debug level.. do you noted it?

Comment (snipe) [reply app/Http/Controllers/Api/SettingsController.php:153]: I did, which is good, but people will generally turn debug on when they're not sure why something else isn't working. Laravel unfortunately fails in mystifying ways when the log file can't be written to, and those issues can be really tough to track down, since there's often not a stack trace that points to any smoking guns.

**Comment by mckaygerhard** [comment]
Also take note those are only parto of the fix and not the final fix, we need to collect information cos i have limited use case over my particular env, there are others reports that those lines will solve the parts that there is no log that helps to us as users to track what happened!

File: app/Http/Controllers/Api/SettingsController.php (hunk 2 above)
Comment (mckaygerhard) [reply app/Http/Controllers/Api/SettingsController.php:153]: I will try to do best for track this king of issues with mail, i used mailtrap in testing env and mailchip in production, thanks for help us.
